modules/ting_covers_rest/src/TingCoversRestService.inc

```diff
@@ -0,0 +1,120 @@
+<?php
+
+use GuzzleHttp\Client;
+use OpenAPI\Client\Configuration;
+use OpenAPI\Client\Api\DefaultApi;
+
+/**
+ * @file
+ * High level implementation of the Ting Covers REST API.
+ */
+class TingCoversRestService {
+  private $config;
+  private $api;
+
+  /**
+   * Construct an API instance.
+   *
+   * @param $host
+   *    The REST service hostname.
+   *
+   * @param bool $token
+   *    The OAuth2 token.
+   *
+   * @param bool $debug
+   *    Enable/disable debugging.
+   */
+  public function __construct($host, $token = FALSE, $debug = FALSE) {
+    $this->config = new Configuration();
+    $this->config->setHost($host)->setDebug($debug);
+    $token ?: $this->config->setAccessToken($token);
+
+    $api_instance = new DefaultApi(
+      new Client(),
+      $this->config
+    );
+
+    $this->api = $api_instance;
+  }
+
+  /**
+   * Returns the configuration object for OpenAPI.
+   *
+   * @return \OpenAPI\Client\Configuration
+   */
+  public function getConfig() {
+    return $this->config;
+  }
+
+  /**
+   * Returns the host string for the requested API.
+   *
+   * @return string
+   *    The hostname for the API.
+   */
+  public function getHost() {
+    return $this->config->getHost();
+  }
+
+  /**
+   * Returns the OAuth token used by the API.
+   *
+   * @return string
+   *    The OAuth token used by the API.
+   */
+  public function getToken() {
+    return $this->config->getAccessToken();
+  }
+
+  /**
+   * Get cover objects by ID type and IDs.
+   *
+   * @param \OpenAPI\Client\Model\Type $type
+   *    The ID type, for example 'pid' or 'issn'.
+   *
+   * @param array $ids
+   *    IDs to lookup of the specified type.
+   *
+   * @return array|\OpenAPI\Client\Model\Cover[]
+   */
+  public function getCovers($type, $ids, $format, $size) {
+    $covers = array();
+    try {
+      $covers = $this->api->coverTypeGet($type, $ids, $format = $format, $generic = true, $size, $limit = null, $offset = 0);
+    }
+    catch (Exception $e) {
+      watchdog_exception('ting_covers_rest', $e);
+    }
+    return $covers;
+  }
+
+  /**
+   * Return covers images from covers, by image format and size.
+   *
+   * @param \OpenAPI\Client\Model\Cover[] $covers
+   *    Array of cover objects to get cover images from.
+   *
+   * @param array $format
+   *    The image formats to return.
+   *
+   * @param array $size
+   *    The image sizes to return.
+   *
+   * @todo
+   *    Get default image sizes from generated swagger code.
+   *
+   * @return array|\OpenAPI\Client\Model\CoverImageUrls[]
```

Comment:
Teknisk set (og hvis det havde været et typehint er `array` korrekt, men til dokumentationsbrug er `\OpenAPI\Client\Model\CoverImageUrls[]` mere præcis. De facto er det jo ikke sådan at der kan returneres to forskellige typer.

Reply:
Bortset fra det er det jo ikke et array ad CoverImageUrls der kommer retur, men et associativt array hvor nøglerne er en string (id) og værdierne er en string (URL).

Reply:
Du har ret. Det fik jeg ikke rettet efter at have lavet om i strukturen. :)

Reply:
Rettet